openapi/generator.go

```diff
@@ -842,7 +842,11 @@ func (g *Generator) enumFromStructField(sf reflect.StructField, fname string, pa
 		values := strings.Split(etag, ",")
 		sftype := sf.Type
 
-		// Use underlying element type if its an array or a slice.
+		// Use pointed element type if it's a pointer
```

Comment:
This condition and the one below can be factored in a single one.

Reply:
I don't think it would work for cases where the type is a pointer AND a slice (`A *[]string`)

There are basically 3 different cases:
- 1st condition is true (type is pointer)
- 2nd condition is true (type is slice)
- both conditions are true (type is pointer to slice)

Reply:
Yeah I get what you say, but actually we're both wrong. What about `**[]string` in this case ? Since we want to get the underlying "scalar" type, we should call `Elem()` until the kind is not one of `Ptr`/`Slice`/`Array`.

That's an interesting case for which we should add more tests while we're at it.